climakitae/explore/agnostic.py

```diff
@@ -614,23 +679,23 @@ def plot_sims(sim_vals, selected_val, time_slice, stats):
     plt.legend()
 
 
-def plot_WRF(sim_vals, metric):
-    """Scatter plot WRF models against their metric values."""
+def plot_WRF(sim_vals, variable, agg_func):
+    """Scatter plot WRF models with their calculated values."""
     sims = [name.split(",")[0] for name in list(sim_vals.simulation.values)]
     sims = [name[4:] for name in sims]
     vals = sim_vals.values
 
     fig, ax = plt.subplots()
     ax.bar(sims, vals)
     ax.set_xlabel("WRF Model, Emission Scenario 3-7.0", labelpad=15, fontsize=12)
-    ax.set_ylabel(f"{metric} ({sim_vals.units})", labelpad=10, fontsize=12)
+    ax.set_ylabel(f"{variable} ({sim_vals.units})", labelpad=10, fontsize=12)
 
     if sim_vals.location_subset == ["coordinate selection"]:
         location = (round(sim_vals.lat.item(), 2), round(sim_vals.lon.item(), 2))

```

Comment:
If `coordinate selection` exists, shouldn't lat/lon exist?

Reply:
I think they should exist in the retrieved data. A culprit is in `agg_lat_lon_sims`.

Reply:
I looked into this behavior, and I believe it's because `area_average` does not return any lat/lon information back into the DataArray about the area at which it was averaged. As a work-around, I added this information back into the DataArray as attributes, and pulled out that information from the results data object in order to plot it later on. 

This fix was made in `_compute_results`, meaning this allows `area_average` to behave properly for both gridcell and area averaging. I think this further motivates the combination of both `agg_lat_lon_sims` and `agg_area_subset_sims` together sooner than later (and redo the inputs to allow a Selections object to be input) in a refactor of the code.